src/Carousel/Carousel.tsx

```diff
@@ -1,64 +1,240 @@
+        const interrupted: React.MutableRefObject<boolean> = React.useRef<boolean>(false);
+
+        /** ----- Utilities ----- */
+
+        const findNewActive: (newNav: NavigationDirection) => number = React.useCallback(
+            (newNav: NavigationDirection): number => {
+                const isInfinite: boolean = infinite || infinite === undefined;
+                switch (newNav) {
+                    case "prev":
+                        return active === 0 ? (isInfinite ? size - 1 : undefined) : active - 1;
+                    case "next":
+                        return active === size - 1 ? (isInfinite ? 0 : undefined) : active + 1;
+                }
+            },
+            [infinite, active, size]
+        );
+
+        /**
+         * Handles navigating to a slide
+         * @param {NavigateTrigger} e Navigation trigger event
+         */
+        const goToSlide: (e: NavigateTrigger, slideTo?: NavigationDirection) => void = React.useCallback(
+            (e: NavigateTrigger, slideTo?: NavigationDirection): void => {
+                e.cancelable && e.preventDefault();
+                let newActive: number;
+                let newNav: NavigationDirection;
+                const target: EventTarget & HTMLElement = e.target as any;
+                if (["mousedown", "touchstart"].some((val) => val === e.type)) {
+                    /** Swipe gesture */
+                    newNav = slideTo;
+                    newActive = findNewActive(newNav);
+                } else {
+                    if (e.type === "click") {
+                        newNav = target.tagName === "LI" ? (Number(target.dataset.slideTo) > active ? "next" : "prev") : (target.dataset.slide as NavigationDirection);
+                    } else {
+                        switch ((e as React.KeyboardEvent<HTMLAnchorElement>).key.toLowerCase()) {
+                            case "arrowleft":
+                                newNav = "prev";
+                                break;
+                            case "arrowright":
+                                newNav = "next";
+                                break;
+                            case "space":
+                            case " ":
+                                newNav = target.dataset.slide as NavigationDirection;
+                        }
+                    }
+                    newActive = target.tagName === "LI" ? Number(target.dataset.slideTo) : findNewActive(newNav);
+                }
+                if ([newNav, newActive].every((val) => val !== undefined)) {
+                    newNav !== nav && setNav(newNav);
+                    newActive !== active && setActive(newActive);
+                }
+            },
+            [active, infinite, nav, findNewActive]
+        );
+
+        const triggerAutoplay: VoidFunction = React.useCallback((): void => {
+            if (autoplay && !interrupted.current) {
+                setTimeout(() => {
```

Comment:
any timeout in the component should be saved and cleared on uinmount in case the component gets destoryed before the autoplayspeed triggers it.

Reply:
added